cores/esp8266/Esp.h

```diff
@@ -98,6 +95,25 @@ class EspClass {
         void deepSleepInstant(uint64_t time_us, RFMode mode = RF_DEFAULT);
         uint64_t deepSleepMax();
 
+        bool forcedModemSleep(uint32_t duration_us = 0, void (*wakeupCb)() = nullptr);
```

Comment:
`using WakeupCb = void(*)()`?

Reply:
There is an existing type for this, see same function in Esp.cpp, plus void(*)() is trivial.

Reply:
general reasoning - this is more verbose. plus, it's repeated twice and it's the same callback both times.
(*mega pedantic note*: `using` is just a code hint aka alias, not something enforcing a specific type as it would with `struct { ... }`. So, just IMO, I'd prefer to have some distinct name rather than re-using something else)

Reply:
But your `using WakeupCb = void(*)();`suggestion relies upon no other definition for `WakeupCb` preexisting in the given namespace or there would a name clash, isn't that right? I respect your opinion, but I am not going to agree .

Reply:
`fpm_wakeup_cb`?

Reply:
Typedefing pointers is considered bad practice in general, even worse for function pointers. Opinions may vary, but I repeat that the function equivalent of 'void*' does not deserve an alias, this is of no utility to anyone. Please stop this.

Reply:
Seeing you probably suggested exposing the Expressif API includes, this is undesirable by breaking the concept of encapsulation/information hiding. For  a void pointer in particular not warranted.